code/game/objects/structures/barsign.dm

```diff
@@ -41,120 +53,448 @@
 	emagged = TRUE
-	req_access = list(ACCESS_SYNDICATE)
+	set_random_sign()
+	req_access = list()
+
+/obj/machinery/barsign/emp_act(severity)
+	if(build_stage >= BARSIGN_WIRED)
+		if(!(stat & EMPED))
+			stat |= EMPED
+			turn_on()
+			set_sign(new /datum/barsign/hiddensigns/empbarsign)
+			playsound(loc, 'sound/effects/sparks4.ogg', 60, TRUE)
+			. = ..()
+
+/obj/machinery/barsign/power_change()
+	. = ..()
+	if(. && (stat & NOPOWER))
+		turn_off()
+
+/obj/machinery/barsign/deconstruct(disassembled=FALSE)
+	if(disassembled)
+		new /obj/item/stack/sheet/metal(drop_location(), 4)
+	else
+		new /obj/item/stack/sheet/metal(drop_location(), 2)
+		if(build_stage >= BARSIGN_WIRED && !(stat & EMPED))
+			new /obj/item/stack/cable_coil(drop_location(), 3)
+		if(build_stage >= BARSIGN_COMPLETE)
+			new /obj/item/shard(drop_location())
+	qdel(src)
+
+
+
+/datum/stack_recipe/barsign_frame
+/datum/stack_recipe/barsign_frame/try_build(mob/user, obj/item/stack/S, multiplier)
+	if(!..())
+		return FALSE
+	var/user_turf = get_turf(user)
+	var/placing_on = get_step(user_turf, user.dir)
+	// Return FALSE if the user isn't facing a wall/window.
+	if(!is_valid_turf(placing_on))
+		to_chat(user, "<span class='warning'>You need to be facing a wall or window to place the [title].</span>")
+		return FALSE
+	// Return FALSE if there isn't space for the entire sign.
+	if((user.dir != NORTH && user.dir != SOUTH) || !is_valid_turf(get_step(placing_on, EAST)))
+		to_chat(user, "<span class='warning'>There is not enough space to place the [title].</span>")
+		return FALSE
+	// Return FALSE if there's already stuff on the wall.
+	if(gotwallitem(user_turf, FLIP_DIR_VERTICALLY(user.dir)) || gotwallitem(get_step(user_turf, EAST), FLIP_DIR_VERTICALLY(user.dir)))
+		to_chat(user, "<span class='warning'>There's already an item on the wall!</span>")
+		return FALSE
+	// Return FALSE if it would cause two bar signs to overlap.
+	if((locate(/obj/machinery/barsign) in get_step(user_turf, WEST)))
+		to_chat(user, "<span class='warning'>There's already a bar sign here!</span>")
+		return FALSE
+	return TRUE
+
+/datum/stack_recipe/barsign_frame/proc/is_valid_turf(turf/T)
+	if(!istype(T))
+		return FALSE
+	if(iswallturf(T))
+		return TRUE
+	if(locate(/obj/structure/window/full) in T)
+		return TRUE
+	return FALSE
+
+/datum/stack_recipe/barsign_frame/post_build(mob/user, obj/item/stack/S, obj/machinery/barsign/O)
+	..()
+	playsound(O, 'sound/items/deconstruct.ogg', 10, TRUE)
+	O.build_stage = BARSIGN_FRAME
+	O.power_state = NO_POWER_USE
+	O.set_light(0)
+	O.update_icon()
+	if(user.dir == SOUTH)
+		O.pixel_y = -32
+	else
+		O.pixel_y = 32
+
+
+
+/obj/item/barsign_electronics
+	name = "bar sign electronics"
+	icon = 'icons/obj/doors/door_assembly.dmi'
+	icon_state = "door_electronics"
+	desc = "A circuit. It has a small data storage component filled with various bar sign designs.\n<span class='notice'>Use a multitool on it to toggle the access restrictions.</span>"
```

Comment:
Move the second part of this description to a examine proc.
And try to move the ```desc =``` next to ```name =``` here and in other places.